client/src/routes/settings/+page.svelte

```diff
@@ -83,25 +93,79 @@
   }
 
   let showDeletionPrompt = false;
-  let errorMsg = "";
+  let showOptions = false;
+  let showPassword = false;
   let errPrompt = false;
+  let linkPass = false;
+
+  let password = "";
+  let errorMsg = "";
+  let googleErr = "";
+  let microsoftErr = "";
+  let passErr = "";
+  let deleteErr = "";
 
+  function getErrorMsg(error: unknown): string {
+    let errorMsg = error instanceof Error ? error.message : String(error);
+    switch (errorMsg) {
+      case "Firebase: Password should be at least 6 characters (auth/weak-password).":
+        return "Password should be at least 6 characters";
+      case "Not Signed in":
+        return errorMsg;
+      case "Firebase: Error (auth/popup-closed-by-user).":
+        return "canceled by user";
+      case "Firebase: Error (auth/email-already-in-use).":
+        return "Email already linked to an account";
+      default:
+        return errorMsg;
+    }
+  }
   function verifyDelete() {
     try {
       deleteAccount();
       // If no error
-      errorMsg = "";
-
-      goto("/");
-      return;
+      window.location.href = "/";
```

Comment:
this doesn't work because you need to await deleteAccount

Reply:
Added await to both functions